pkg/function/restore_data_using_kopia_server.go

```diff
@@ -0,0 +1,261 @@
+		cli,
+		hostname,
+		namespace,
+		restorePath,
+		serverAddress,
+		fingerprint,
+		snapID,
+		username,
+		userPassphrase,
+		sparseRestore,
+	)
+	return pr.Run(ctx, podFunc)
+}
+
+func restoreDataFromServerPodFunc(
+	cli kubernetes.Interface,
+	hostname,
+	namespace,
+	restorePath,
+	serverAddress,
+	fingerprint,
+	snapID,
+	username,
+	userPassphrase string,
+	sparseRestore bool,
+) func(ctx context.Context, pod *corev1.Pod) (map[string]any, error) {
+	return func(ctx context.Context, pod *corev1.Pod) (map[string]any, error) {
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrap(err, "Failed while waiting for Pod: "+pod.Name+" to be ready")
+		}
+
+		contentCacheMB, metadataCacheMB := kopiacmd.GetCacheSizeSettingsForSnapshot()
+		configFile, logDirectory := kankopia.CustomConfigFileAndLogDirectory(hostname)
+
+		cmd := kopiacmd.RepositoryConnectServerCommand(
+			kopiacmd.RepositoryServerCommandArgs{
+				UserPassword:    userPassphrase,
+				ConfigFilePath:  configFile,
+				LogDirectory:    logDirectory,
+				CacheDirectory:  kopiacmd.DefaultCacheDirectory,
+				Hostname:        hostname,
+				ServerURL:       serverAddress,
+				Fingerprint:     fingerprint,
+				Username:        username,
+				ContentCacheMB:  contentCacheMB,
+				MetadataCacheMB: metadataCacheMB,
+			})
+		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stdout)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stderr)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to connect to Kopia API server")
+		}
+
+		cmd = kopiacmd.SnapshotRestore(
+			kopiacmd.SnapshotRestoreCommandArgs{
+				CommandArgs: &kopiacmd.CommandArgs{
+					RepoPassword:   "",
+					ConfigFilePath: configFile,
+					LogDirectory:   logDirectory,
+				},
+				SnapID:                 snapID,
+				TargetPath:             restorePath,
+				SparseRestore:          sparseRestore,
+				IgnorePermissionErrors: true,
+			})
+		stdout, stderr, err = kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stdout)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stderr)
+		return nil, errors.Wrap(err, "Failed to restore backup from Kopia API server")
+	}
+}
+
+func validateAndGetOptArgsForRestore(tp param.TemplateParams, args map[string]any) (pod string, vols map[string]string, podOverride crv1alpha1.JSONMap, err error) {
+	if err = OptArg(args, RestoreDataPodArg, &pod, ""); err != nil {
+		return pod, vols, podOverride, err
+	}
+	if err = OptArg(args, RestoreDataVolsArg, &vols, nil); err != nil {
+		return pod, vols, podOverride, err
+	}
+	if (pod != "") && (len(vols) > 0) {
+		return pod, vols, podOverride, errors.New(fmt.Sprintf("Exactly one of the %s or %s" arguments are required, but both are provided, RestoreDataPodArg, RestoreDataVolsArg))
+	}
```

Comment:
What is the expected behaviour if both the args are not set?

Reply:
The function should be erroring out with the mentioned error